lib/ssz_ex.ex

```diff
@@ -53,13 +56,17 @@ defmodule LambdaEthereumConsensus.SszEx do
   def decode(binary, {:int, size}), do: decode_uint(binary, size)
   def decode(value, {:bytes, _}), do: {:ok, value}
 
-  def decode(binary, {:list, basic_type, size}) do
+  def decode(binary, {:list, basic_type, size} = schema) do
     if variable_size?(basic_type),
       do: decode_variable_list(binary, basic_type, size),
-      else: decode_list(binary, basic_type, size)
+      else: decode_list(binary, basic_type, size, schema)
```

Comment:
The `decode_list` is a very confusing function, most likely because:

1. It's called `list` but it's used for both list and vector. It should be called `decode_collection`, or something like that. 
2. It passes the whole schema instead of just passing :list or :vector.

We will delete this function. In general it's very weird to pass an atom like :list or vector UNLESS there will be a dynamic or unpredictable situation for that parameter. Here, we'll always pass :list when it's a list or :vector when it's a vector. This is very similar to what I commented on assert_ssz_valid vs assert_ssz_invalid.

As a matter of fact, this is over-reutilization makes the code exceedenly complex, as the only thing these functions share is getting the fixed size of a variable and then the decoding sections. All checks are different, or may not even exist.

Let's have `decode_fixed_list` and `decode_fixed_vector` functions. I'll comment below some tips on how to do that.